policyengine_api/ai_prompts/simulation_analysis_prompt.py

```diff
@@ -0,0 +1,131 @@
+from policyengine_api.ai_templates.simulation_analysis_template import (
+    simulation_analysis_template,
+    audience_descriptions,
+)
+from pydantic import BaseModel
+from typing import Any, Annotated
+import json
+
+
+class InboundParameters(BaseModel):
+    currency: str
+    country_id: str
+    selected_version: str
+    time_period: str
+    impact: dict[str, dict[str, Any]]
+    policy_label: str
+    policy: dict[str, Any]
+    region: str
+    relevant_parameter_baseline_values: list[dict[str, Any]]
+    relevant_parameters: list[dict[str, Any]]
+    audience: str
+
+
+class AllParameters(InboundParameters):
+    model_config = {"exclude": {"audience"}}
+    enhanced_cps_template: str
+    dialect: str
+    data_source: str
+    poverty_measure: str
+    poverty_rate_change_text: str
+    poverty_by_race_text: str
+    audience_description: str
+    country_id_uppercase: Annotated[str, "Uppercase two-letter country ID"]
+    impact_budget: Annotated[str, "JSON deserialized to string"]
+    impact_intra_decile: Annotated[str, "JSON deserialized to string"]
+    impact_decile: Annotated[str, "JSON deserialized to string"]
+    impact_inequality: Annotated[str, "JSON deserialized to string"]
+    impact_poverty: Annotated[str, "JSON deserialized to string"]
+    impact_deep_poverty: Annotated[str, "JSON deserialized to string"]
+    impact_poverty_by_gender: Annotated[str, "JSON deserialized to string"]
+
+
+def generate_simulation_analysis_prompt(params: InboundParameters) -> str:
+    """
+    Generate AI prompt for economy-wide simulations
+    """
+
+    parameters: InboundParameters = InboundParameters.model_validate(params)
+
+    enhanced_cps_template: str = (
+        """Explicitly mention that this analysis uses PolicyEngine Enhanced CPS, constructed 
+    from the 2022 Current Population Survey and the 2015 IRS Public Use File, and calibrated 
```

Comment:
Could we pull this in from dataset metadata? If not easily here could you file an issue?
```suggestion
    from the 2023 Current Population Survey and the 2015 IRS Public Use File, and calibrated 
```

Reply:
We cannot. In fact, if we move to the implementation of datasets that we've been considering so far, the dataset would only be a raw data file pulled from Hugging Face and/or GitHub. We'd need to consider either building some tools and norms around storing that information within Hugging Face and accessing via API, or building some form of light wrapper class over the data, which I believe Nikhil is against.

Reply:
Thanks for the context and filing https://github.com/PolicyEngine/policyengine-api/issues/2174